docs/features/compilation-from-portable-pdb.md

```diff
@@ -0,0 +1,215 @@
+## compilation-from-portable-pdb
+
+Compilation from portable PDBs today is not completely possible, but is desirable in order to help reconstruct a compilation from source provided via source link or embedded in a pdb. Motivation is derived from [roslyn 41395](https://github.com/dotnet/roslyn/issues/41395).
```

Comment:
@ryzngard Could you spell out an end-to-end scenario? In particular, if this information is embedded in a PDB, how would one get the information out? That should probably be useful to document here.

Reply:
Agreed, we should mention the information is available in CustomDebugInformation table under specific GUIDs. 
Once we finalize the design I also plan to update the [Portable PDB spec](https://github.com/dotnet/runtime/blob/master/docs/design/specs/PortablePdb-Metadata.md#CustomDebugInformationTable) to list these new CDIs. We should also update mdv (MetadataVisualizer) and Pdb2Xml to recognize these.

Reply:
Added examples of how to get information out of the pdb. Still writing more information for end-to-end scenario

Reply:
#44703 tracks end-to-end integration test for us. I added a note in the markdown file that scenario should better be explained once that's done, next to a sentence with the overarching goal